src/kbmod/reprojection.py

```diff
@@ -40,11 +40,24 @@ def reproject_image(image, original_wcs, common_wcs):
     image_data = CCDData(image, unit="adu")
     image_data.wcs = original_wcs
 
-    new_image, footprint = reproject.reproject_adaptive(
-        image_data, common_wcs, shape_out=common_wcs.array_shape, bad_value_mode="ignore"
-    )
+    footprint = np.zeros(common_wcs.array_shape, dtype=np.ubyte)
+
+    # if the input image is actually a stack of images, we need to duplicate the
+    # footprint to match the total number of images.
+    if type(image) is list:
+        footprint = np.repeat(footprint[np.newaxis, :, :], len(image), axis=0)
+
+    new_image, _ = reproject.reproject_adaptive(
+        image_data, common_wcs, shape_out=common_wcs.array_shape, bad_value_mode="ignore",
+        output_footprint=footprint
+        )
+
+    # if we passed in a stack of ndarrays (i.e. science, varianace, mask), we only
+    # need to return the first footprint, as they should all be the same.
+    if footprint.ndim == 3:
+        footprint = footprint[0]
 
-    return new_image, footprint
+    return new_image.astype(np.float32), footprint
```

Comment:
Returning `new_image` as float32. This will create a copy in memory from the float64 to the float32 version, but ultimately the goal is to reduce the amount of time that a float64 array is sitting in memory.